ocs_ci/deployment/disconnected.py

```diff
@@ -302,37 +381,80 @@ def mirror_index_image_via_oc_mirror(index_image, packages, icsp=None):
 
-    # create ImageContentSourcePolicy
-    icsp_file = os.path.join(
-        f"{mirroring_manifests_dir}",
-        "imageContentSourcePolicy.yaml",
-    )
-    # make icsp name unique - append run_id
-    with open(icsp_file) as f:
-        icsp_content = yaml.safe_load(f)
-    icsp_content["metadata"]["name"] = f"odf-{config.RUN['run_id']}"
-    with open(icsp_file, "w") as f:
-        yaml.dump(icsp_content, f)
-    exec_cmd(f"oc apply -f {icsp_file}")
-    wait_for_machineconfigpool_status("all")
+    if idms:
+        cmd += " --dry-run"
+        # running the command again with --dry-run to get mapping file
+        # so that we can pass it and get the failed mirrors
+        exec_cmd(cmd, timeout=18000)
+        mapping_file = os.path.join(
+            f"{mirroring_manifests_dir}",
+            "working-dir/dry-run/mapping.txt",
+        )
+        with open(mapping_file, "r") as file:
+            lines = file.readlines()
 
-    # get mirrored index image url from prepared catalogSource file
-    cs_file = glob.glob(
-        os.path.join(
+        # Remove 'docker://' from each line
+        updated_lines = [line.replace("docker://", "") for line in lines]
+
+        with open(mapping_file, "w") as file:
+            file.writelines(updated_lines)
+
+        mirror_images_from_mapping_file(mapping_file, idms=idms)
+
+        # # create ImageContentSourcePolicy
+        # icsp_file = os.path.join(
+        #     f"{mirroring_manifests_dir}",
+        #     "imageContentSourcePolicy.yaml",
+        # )
+
+        # create ImageDigestMirrorSet
+        idms_file = os.path.join(
             f"{mirroring_manifests_dir}",
-            "catalogSource-*.yaml",
+            "working-dir/cluster-resources/idms-oc-mirror.yaml",
+        )
+
+        # # make icsp name unique - append run_id
+        # with open(icsp_file) as f:
+        #     icsp_content = yaml.safe_load(f)
+        # icsp_content["metadata"]["name"] = f"odf-{config.RUN['run_id']}"
+        # with open(icsp_file, "w") as f:
+        #     yaml.dump(icsp_content, f)
+        # exec_cmd(f"oc apply -f {icsp_file}")
+        # wait_for_machineconfigpool_status("all")
+
+        # make idms name unique - append run_id
+        with open(idms_file) as f:
+            idms_content = yaml.safe_load(f)
+        idms_content["metadata"]["name"] = f"odf-{config.RUN['run_id']}"
+        with open(idms_file, "w") as f:
+            yaml.dump(idms_content, f)
+        exec_cmd(f"oc apply -f {idms_file}")
+        wait_for_machineconfigpool_status("all")
+
+    # get mirrored index image url from prepared catalogSource file
+    if idms:
+        cs_file = glob.glob(
+            os.path.join(
+                f"{mirroring_manifests_dir}",
+                "working-dir/cluster-resources/cs*.yaml",
+            )
+        )
+    else:
+        cs_file = glob.glob(
+            os.path.join(
+                f"{mirroring_manifests_dir}",
+                "cs-*.yaml",
+            )
```

Comment:
As mentioned above, the usage of `oc mirror --v2` should be differentiated based on OCP version, not based on usage of `idms` parameter, so this condition should be also based on OCP version not based on the `idms` parameter. (Mainly because for GA version neither `icsp` neither `idms` parameter is used and we still want to use `oc mirror --v2` for OCP 4.18 and above.